src/memtest.rs

```diff
@@ -55,6 +60,29 @@ pub enum MemtestKind {
     BlockSeq,
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct ParseMemtestKindError;
+
+impl MemtestKind {
+    pub fn all_test_kinds() -> Vec<Self> {
```

Comment:
nit: I would probably just call this `all`, since the type itself already has the words "test" and "kind" in it.

Also, instead of a function, it can easily be made into an associated `const`. That is faster and more flexible, as you can iterate it without heap-allocating a `Vec` first.

```Rust
pub const ALL: &[Self] = &[
    Self::OwnAddressBasic,
    // ...
];
```